common/lib/xmodule/xmodule/js/src/word_cloud/word_cloud_main.js

```diff
@@ -154,136 +158,188 @@
-            .end()
-            .find('.total_num_words').html(response.total_count);
+                .addClass('active');
+
+            HtmlUtils.setHtml(
+                cloudSectionEl.find('.your_words'),
+                HtmlUtils.HTML(studentWordsStr)
+            );
+
+            HtmlUtils.setHtml(
+                cloudSectionEl.find('.your_words').end().find('.total_num_words'),
+                HtmlUtils.interpolateHtml(
+                    gettext('{start_strong}{total}{end_strong} words submitted in total.'),
+                    {
+                        start_strong: HtmlUtils.HTML('<strong>'),
+                        end_strong: HtmlUtils.HTML('</strong>'),
+                        total: response.total_count
+                    }
+                )
+            );
 
             $(cloudSectionEl.attr('id') + ' .word_cloud').empty();
 
-        // Actual drawing of word cloud.
+            // Actual drawing of word cloud.
             groupEl = d3.select('#' + cloudSectionEl.attr('id') + ' .word_cloud').append('svg')
-                .attr('width', this.width)
-                .attr('height', this.height)
-                .append('g')
-                .attr('transform', 'translate(' + (0.5 * this.width) + ',' + (0.5 * this.height) + ')')
-                .selectAll('text')
-                .data(words)
-                .enter().append('g');
+                    .attr('width', this.width)
+                    .attr('height', this.height)
+                    .append('g')
+                    .attr('transform', 'translate(' + (0.5 * this.width) + ',' + (0.5 * this.height) + ')')
+                    .selectAll('text')
+                    .data(words)
+                    .enter()
+                    .append('g')
+                    .attr('data-id', function() {
+                        return Math.floor(Math.random() * (999 - 1 + 1)) + 1;
+                    })
+                    .attr('aria-describedby', function() {
+                        return HtmlUtils.interpolateHtml(
+                            gettext('text_word_{word} title_word_{word}'),
+                            {
+                                word: generateUniqueId(cloudSectionEl.attr('id'), $(this).data('id'))
+                            }
+                        );
+                    });
 
             groupEl
-            .append('title')
-            .text(function(d) {
-                var res = '';
-
-                $.each(response.top_words, function(index, value) {
-                    if (value.text === d.text) {
-                        res = value.percent + '%';
-
-                        return;
-                    }
+                .append('title')
+                .attr('id', function() {
+                    return HtmlUtils.interpolateHtml(
+                        gettext('title_word_{word}'),
+                        {
+                            word: generateUniqueId(cloudSectionEl.attr('id'), $(this).parent().data('id'))
```

Comment:
I'm surprised that "this" works in this function callback. Happy, but surprised.